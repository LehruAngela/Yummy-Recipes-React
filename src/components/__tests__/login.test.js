import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import Login from '../login';

describe('Login', () => {
    const wrapper = shallow(<Login />);
    const preventDefault = jest.fn();

    it('renders properly', () => {
        expect(shallowToJson(wrapper)).toMatchSnapshot();
        }); 

    it('handles Login', () => {
        expect(wrapper.instance().handleLogin({ preventDefault }));
        });
})