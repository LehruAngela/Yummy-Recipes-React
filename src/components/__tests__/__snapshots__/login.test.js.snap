// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login renders properly 1`] = `
<div
  className="form-margin"
>
  <h1>
    Login to your account
  </h1>
  <br />
  <form
    onSubmit={[Function]}
  >
    <div
      className="form-group"
    >
      <h4>
        <label
          for="email"
        >
          Email:
        </label>
      </h4>
      <input
        name="email"
        onChange={[Function]}
        type="email"
        value=""
      />
    </div>
    <br />
    <div
      className="form-group"
    >
      <h4>
        <label
          for="password"
        >
          Password:
        </label>
      </h4>
      <input
        name="password"
        onChange={[Function]}
        type="password"
        value=""
      />
    </div>
    <div
      className="form-group"
    >
      <button
        className="btn btn-outline-warning"
        type="submit"
      >
        Login
      </button>
    </div>
    <div>
      <a
        data-target="#exampleModal"
        data-toggle="modal"
        href="#"
      >
        Forgotten Password?
      </a>
      <SendEmail />
    </div>
    <br />
    <br />
  </form>
  <div>
    <h4>
      Don't have an account?
    </h4>
    <Link
      replace={false}
      to="/"
    >
      <button
        className="btn btn-outline-warning"
        type="submit"
      >
        Register
      </button>
    </Link>
  </div>
  <br />
  <br />
  <footer>
    <p
      class="copyright text-muted small footer"
    >
      Copyright Â© Gela 2018.
    </p>
  </footer>
</div>
`;
